### PR TITLE
[WIP]feat: add notif and email for pending to expired orders after 30 minutes

### LINKED ISSUE #6078: Send notif and email on pending -> expired order transition
This is a sub issue
Parent issue: #6072

When the ticket reservation expires after 30 minutes, send an email and notification to the user: 
"Do you want to order this ticket still? You can order through this link within 24 hours ( if ticket is available )"

### PR DESCRIPTION
**Pull request by uds5501**
Fixes #6078 
Related to #6072 

#### Changes proposed in this pull request:

- Sends an email and notification when pending order is expired after 30 minutes

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by uds5501** [comment]
@shreyanshdwivedi @mrsaicharan1 Please take a look at the sent messages in notification and email and suggest any edits, if required

File: app/api/helpers/system_mails.py [hunk 1]
```diff
@@ -261,5 +261,15 @@
         'message': (
             u"This is a  <strong> Test </strong> E-mail."
         )
+    },
+    PENDING_ORDER_EXPIRED: {
```
Comment (shreyanshdwivedi) [inline app/api/helpers/system_mails.py]: @uds5501 this issue is supposed to send an email and notification to user using which he/she can deactivate the expired order. So this issue involves an endpoint maybe where user can access his/her expired ticket within 24 hours and reactivate it and complete order by paying.
So maybe you can proceed by understanding these points -
 - User clicks on the link and is redirected to a page of his/her expired ticket (within 24 hours). 
 - The status is changed from `expired` to `pending` when user uses the link
 - The user can pay using the payment methods available on the `pending order page`.
 - If user fails to complete payment 2nd time, the ticket should be expired and then this normal mail should be sent out to him that he/she can order by visiting tickets page of the event

Comment (uds5501) [reply app/api/helpers/system_mails.py]: @shreyanshdwivedi Okay, So here are a few things which I will add, please let me know if they confirm to the pattern required in this refactor.

- The expiry mail will only be sent if there are any tickets left for that particular event.
- If user uses the link, the order status will be set to `pending`



> * If user fails to complete payment 2nd time, the ticket should be expired and then this normal mail should be sent out to him that he/she can order by visiting tickets page of the event

I am just no sure how to implement this. Any ideas?

Comment (shreyanshdwivedi) [reply app/api/helpers/system_mails.py]: Corresponding to your first point - As user has pending order, the tickets are already reseved. No need to check if ticket exists while sending out mail for expired order.
When a user's pending ticket expires normally send him a mail with a link using which he/she can access that expired ticket and update it's status to pending again.
You can take example of reset password where email is send out alongwith a link and user accesses a page via link to update password. Try to create a endpoint to handle the click on link

Comment (uds5501) [reply app/api/helpers/system_mails.py]: @shreyanshdwivedi Understood.

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi it's the same thing actually. Expiry time is 30 mins and
there's a cron job which runs every 45 min.
15 mins is out for form filling. Then the user has 30mins to complete
payment. So it's alright!

**Comment by uds5501** [comment]
@mrsaicharan1 @shreyanshdwivedi The link needs to be updated with the frontend route. So maybe will keep it in WIP status for now while we figure out how to make the frontend work and continue adding corresponding checks. Sounds Good?

**Pull request closed without merging**